File: src/main/java/org/zibble/discordmessenger/components/messagable/MentionType.java
package org.zibble.discordmessenger.components.messagable;

import com.google.gson.JsonObject;
import com.google.gson.JsonPrimitive;
import org.zibble.discordmessenger.components.JsonSerializable;

public enum MentionType implements JsonSerializable {

    USER,
    ROLE,
    CHANNEL,
    EMOTE,
    HERE,
    EVERYONE,
    ;

    public static MentionType fromJson(JsonObject json) {
        return MentionType.valueOf(json.getAsJsonPrimitive().getAsString());
    }

    @Override
    public JsonObject toJson() {
        return new JsonPrimitive(this.name()).getAsJsonObject();
    }

    @Override
    public String toJsonString() {
        return this.toJson().toString();
    }

}

File: src/main/kotlin/org/zibble/discordmessenger/components/messagable/MentionType.kt
package org.zibble.discordmessenger.components.messagable

import org.zibble.discordmessenger.components.JsonSerializable

enum class MentionType : JsonSerializable {

    USER,
    ROLE,
    CHANNEL,
    EMOTE,
    HERE,
    EVERYONE,
    ;

}